[docs][account-types]: Update the outdated information (#18552)
expo/expo: docs/pages/accounts/account-types.md

<ImageSpotlight alt="Transfer Projects settings panel." src="/static/images/accounts/transfer-project.jpg" style={{maxWidth: 720}} />

#### Caveats

Before transferring a project, make sure that it is on Expo SDK 43 or above. If not, you must upgrade it and rebuild it for OTA updates to continue to work on that project.

> **Note**: If you want to transfer the ownership of a project from your Personal or Organization Account (source) to another person or company (destination), and you cannot be given "Owner or "Admin" permissions on the destination account, you can create an escrow account (a new Organization Account). This solves the problem that a user must be an "Owner" or "Admin" on both source and destination accounts to transfer projects between them. Once the escrow account is created, you can grant the ultimate destination account member the Owner role on the escrow account and safely transfer the project to the escrow account. The receiving person or company can then transfer it to their destination account from the escrow account without having had access to the destination account itself.

## Teams

> > **Note**: For all new accounts where you want to share access, we recommend you to [create a new Organization](#organizations) Account. It allows more flexibility in role assignments Creating a Team Account is deprecated.

Reviewer: missing period

- You can continue to deliver updates and push notifications to your users.
- You will still be able to use any iOS or Android credentials stored on Expo's servers.
- Any integrations using your personal access token or webhooks will continue to operate and are transferred to the new designated owner.
- Your EAS subscription will continue without interruption.
- Your production apps will continue to operate without interruption.

The Organization Account will adopt your old username. Converting a Personal Account into an Organization Account requires that you designate a new user as the owner of the Organization account. You will have the opportunity to designate an existing user or sign up as a new user during the account conversion process.

The example below is Step 3 in the conversion process on a Personal Account with the username "kamalaya8":

<ImageSpotlight alt="Example demonstrating the Step 3 of the conversion process." src="/static/images/accounts/converting-personal-account.jpg" style={{maxWidth: 720}} />

> After completing the conversion process, you can no longer log in as the old Personal Account's user. To continue using Expo services, you will log into [expo.dev](https://expo.dev/) with the user you selected or created to manage the Organization Account. Then, select the organization from the top left dropdown to access the organization.

Reviewer 2: ```suggestion
After completing the conversion process, you will no longer be able to log in as the your old user. To continue using Expo services, log into [expo.dev](https://expo.dev/) with the user you selected or created during the conversion process. Then, select the organization from the top left dropdown to access the organization.
```

### Inviting members

> Since the Organization is a shared container with your team, you can invite other members to it. To invite a new member:

Reviewer 2: ```suggestion
Other Expo users can be invited to join your Organization. To invite a new member:
```

- From the Organization's dashboard, select Organization settings from the sidebar. This will open all the settings available for the organization.
- In Account settings, select Members from the sidebar.
- Under Members, click the button "Invite Member". This will open a form to invite a member to the organization.

> - In the form, enter the email of the member you want to invite and select their role. For more information, see [role privileges](/accounts/working-together/#managing-access).

Reviewer 2: ```suggestion
- In the form, enter the email of the user you want to invite and select the role they should have upon joining the organization. For more information, see [role privileges](/accounts/working-together/#managing-access).
```

<ImageSpotlight alt="Example demonstrating multiple members in an organization." src="/static/images/accounts/members.jpg" style={{maxWidth: 720}} />

When inviting a new member, keep in mind:

> - Only a member with an Admin or an Owner role can invite others.

Reviewer 2: ```suggestion
- Only members with an Admin or an Owner role can invite others.
```

> - Member with an Owner role can give members/invitees any role.

Reviewer 2: ```suggestion
- Members with an Owner role can grant members and invitees any role.
```

> - Member with an Admin role can only give members/invitees up to and including Admin role.

Reviewer 2: ```suggestion
- Members with an Admin role can only give members and invitees up to and including Admin role (every role but Owner).
```

### Renaming an Account

> Accounts can be renamed a limited number of times. Only Owners can rename accounts. Visit [the account settings](https://expo.dev/accounts/[account]/settings) and follow the steps under [Rename Account](https://expo.dev/settings#rename-account).

Reviewer 2: ```suggestion
Accounts can be renamed a limited number of times. Only Owners can rename accounts. To rename an account, visit [the account settings](https://expo.dev/accounts/[account]/settings) and follow the steps under [Rename Account](https://expo.dev/accounts/[account]/settings#rename-account).
```

<ImageSpotlight alt="Rename Account settings panel." src="/static/images/accounts/rename-account.jpg" style={{maxWidth: 720}} />

#### Caveats

After renaming an account, new publishes to projects within the account must be on Expo SDK 43 or above.

### Transferring projects between Accounts

Projects can be transferred a limited number of times. A user must be an Owner or Admin on both source and destination accounts to transfer projects between them. Visit [the project settings page](https://expo.dev/accounts/[account]/projects/[project]/settings) and follow the steps under Transfer project.

<ImageSpotlight alt="Transfer Projects settings panel." src="/static/images/accounts/transfer-project.jpg" style={{maxWidth: 720}} />

#### Caveats

> Before transferring a project, make sure that your it is on Expo SDK 43 or above. If not, you must upgrade it and rebuild it for OTA updates to continue to work on that project.

Reviewer 2: ```suggestion
Before transferring a project, make sure that it is on Expo SDK 43 or above. If not, you must upgrade it and rebuild it for OTA updates to continue to work on that project.
```

title: Account Types
---

import ImageSpotlight from '~/components/plugins/ImageSpotlight';

An Expo account is a container that holds Expo projects and allows for different amounts of collaboration. There are three types of Expo accounts; Personal, Team, and Organization. What type of account you choose to put a new project in depends on the nature of the project and, if you plan to collaborate with others, with whom you plan to collaborate.

When you [sign up for an account](https://expo.dev/signup) with Expo, a Personal Account is automatically created for you. If you want multiple members to have access to project(s), you can create an Organization Account.

> Before Organization accounts, the only way to work in a group was to add members directly to a Personal Account. This was called a Team Account. Only the original member can have an Owner role in a Team Account. An Organization Account can have multiple members with the Owner role. You can choose whichever account type that makes the most sense for your group. Both group account types allow the following:

Author: > move the history of team accounts and other team account info to the team account section

I've removed the history of team accounts from the introduction. I do not think its relevant. Let me know if you think it should be there.

>move the organization account info to the organization account section

Re-wrote Organization section introduction which includes the new info in detail.

>as for the role stuff, I think it's fine to just link to the working-together page and not get into too much detail here. We may need to update that page though, idk.

Yeah, I agree on, that's what I've done. I did open an issue in Linear (https://linear.app/expo/issue/ENG-5975/verify-and-update-outdated-info-on-roles-privileges) to update that page and have some questions. When you have time, can you please take a look? 🤗

- Sharing projects
- Assigning member permissions (only one Owner in a Team Account)
- Sharing an [EAS Subscription](https://docs.expo.dev/eas/)

Original or added Owners can invite and assign members of their group a role. This role determines what actions they can or can not perform. Four types of roles are available:

- Owner
- Admin
- Developer
- Viewer

Each role has different privileges. Learn more about [role privileges](/accounts/working-together/#managing-access).

> > **Note**: You can switch between accounts or create a new Organization Account from the dropdown in the top left corner. If you want to convert your Personal Account to an Organization Account, select the avatar from the top right corner of the dashboard and then select User Settings.

Author: The section `Converting Personal Accounts into Organizations` already includes these instructions so removed the note from into.

## Personal Accounts

When you sign up a Personal Account is automatically created for you. This account is a good place to work on your personal projects.

> > **Note**: You must not share authentication credentials for your Personal Account with anyone for any reason. Members can log in to their Personal Accounts to access any Team or Organization they are part of. Access to these accounts is available from the dropdown in the top left corner of the dashboard on [expo.dev](https://expo.dev).

Author: Removed the second part and I think in each Team and Organization section talks about accessing them in detail in their own section. So removed it altogether.

## Teams

> > **Note**: For all new accounts where you want to share access, we recommend you [create a new Organization](#organizations) Account. It allows more flexibility in role assignments and allows you to keep some projects private to you. However, a Team Account is still a viable way to give other users access to all your projects, just keep in mind that granting someone access to your account grants them access to all of your personal projects.

Author: Updated the first part of the note as per your suggestion (that is to remove "and allows you to keep some projects private to you").

Replaced the second part " However, a Team Account is still a viable way to give other users access to all your personal projects, just keep in mind that granting someone access to your Personal account grants them access to all of your personal projects." with "Creating a Team Account is deprecated."

title: Account Types
---

import ImageSpotlight from '~/components/plugins/ImageSpotlight';

> There are three types of Expo accounts; Personal, Team, and Organization. The type you choose depends on whether you work alone on projects or in a group. However, you must start with at least one Personal Account to use Expo.

Reviewer: I advised that we still include Team Accounts in the documentation for now since [this discussion](https://exponent-internal.slack.com/archives/CTR08334G/p1659653015542189) is still pending

> There are three types of Expo accounts; Personal, Team, and Organization. The type you choose depends on whether you work alone on projects or in a group. However, you must start with at least one Personal Account to use Expo.

Author: We'll have to conclude internally as to whether include Team related info in docs or not. As Kelley mentioned, I followed her suggestion.

My suggestion, is to keep the info for now since its still a feature available but remove it as soon as Team accounts functionality are disabled.

When you [sign up for an account](https://expo.dev/signup) with Expo, you are creating a Personal Account. If you want multiple members to have access to project(s), you can create an Organization Account from within your Personal Account and transfer projects individually to the Organization account, if required. Another option is to convert your Personal Account directly to an Organization Account if you want all existing projects transferred automatically.

> Before Organization accounts, the only way to work in a group was to add members directly to a Personal Account This is called a Team Account. Only the original member can have an Owner role in a Team Account. An Organization Account can have multiple members with the Owner role. You can choose whichever account type that makes the most sense for your group. Both group account types allow the following:

Author: I'll move the Team Account at the end of the page (please see my comment: https://github.com/expo/expo/pull/18552/files#r942451815) but I am also open to remove it completely from the page or updating the section title as:

```markdown
## Team (legacy).
```

There is a note in the Team account section that directs the user to Organization. here is the note:

```markdown
> **Note**: For all new accounts where you want to share access, we recommend you [create a new Organization](#organizations) Account. It allows more flexibility in role assignments and allows you to keep some projects private to you. However, a Team Account is still a viable way to give other users access to all your projects, just keep in mind that no projects in your account can be private to you if you make a Team Account unless you remove all members.
```

- Sharing projects
- Assigning member permissions (only one Owner in a Team Account)
- Sharing an [EAS Subscription](https://docs.expo.dev/eas/)

> Original or added Owners can invite and assign members of their group a role. This role determines what actions they can or can not perform. Four types of roles are available:

Author: In the latest PR, I'm removing this part and merging the bullet points above in the organization section. After removing the team (and deprecating it at the bottom of the page), I do not see a point in explaining "group" accounts here separately.

- Owner
- Admin
- Developer
- Viewer

Each role has different privileges. Learn more about [role privileges](/accounts/working-together/#managing-access).

> **Note**: You can switch between accounts or create a new Organization Account from the dropdown in the top left corner. If you want to convert your Personal Account to an Organization Account, select the avatar from the top right corner of the dashboard and then select User Settings.

## Personal Accounts

When you sign up for an account with Expo, you are creating a Personal Account. All Expo users have their own Personal Account(s). Expo uses it to identify you as an individual safely.

> A Personal Account is the default account type. For example, your Personal Account username is used when you use the [contact form](https://expo.dev/contact) to reach out for support when you are logged in.

Reviewer: I think it's sufficient to remove this line.

> **Note**: You must not share authentication credentials for your Personal Account with anyone for any reason. Members can log in to their Personal Accounts to access any Team or Organization they are part of. Access to these accounts is available from the dropdown in the top left corner of the dashboard on [expo.dev](https://expo.dev).

## Teams

> **Note**: For all new accounts where you want to share access, we recommend you [create a new Organization](#organizations) Account. It allows more flexibility in role assignments and allows you to keep some projects private to you. However, a Team Account is still a viable way to give other users access to all your projects, just keep in mind that no projects in your account can be private to you if you make a Team Account unless you remove all members.

> To create a Team Account, simply [invite Team members](/accounts/working-together/#adding-members) to a Personal Account. An email will be sent that prompts them to create a Personal Expo Account if they do not already have one. In a Team Account, the team's creator has the Owner role only.

Author: The changes you suggested sound confusing to me. How about:

```suggestion
To create a Team Account, simply [invite Team members](/accounts/working-together/#adding-members) to a Personal Account. An email will be sent that prompts them to create a Personal Expo Account if they do not already have one. Only the person (for whom a Personal Account was automatically created) is the Owner of that Personal Account and, therefore, the Team Account.
```

For members to access the Team projects, they select the Owner’s account name from the dropdown in the top left under "Teams". The Owner can login to their Personal Account and manage it as usual.

## Organizations

> An Organization Account is useful for creating a project that may outlive your involvement. If you are working on a shared project with other members of a company or a group of developers, we recommend you [create an Organization](expo.dev/create-organization) Account.

Author: Thanks for the suggestion! I'll update this section separately, I think I've got the idea of what is required here.

It is useful to have Organization when:

- Transferring control of a project without losing access to your Personal Account
- Assigning more than one Owner
- Isolating expenses for work-related projects for submitting expenses
- Structuring projects for different contexts. For example, working for different clients

### Creating new Organizations

> To create a new Organization, log in to your Personal Account and [open the link](https://expo.dev/create-organization) to create a new organization.

Author: Didn't know about the difference here. Thanks!

> Alternatively, if you are logged in to your Personal Account, you can create a new Organization from the dashboard:

Author: In that case, I'll remove the previous changes and stick to this version.

### Converting Personal Accounts into Organizations

You can convert your Personal Account into an Organization when you want to share access to projects with other members and assign each member a role-based privilege.

From the User Settings of your Personal Account, go to [Convert your account into an organization](https://expo.dev/settings#convert-account) section to start the process.

When you are going through this process, we take a lot of care to make sure that all of the functionality that you and your users rely on will continue to work as expected:

- You can continue to deliver updates and push notifications to your users.
- You will still be able to use any iOS or Android credentials stored on Expo's servers.
- Any integrations using your personal access token or webhooks will continue to operate and are transferred to the new user designated.
- Your EAS subscription will continue without interruption.
- Your production apps will continue to operate without interruption.

> The Organization Account will adopt your old username. Converting a Personal Account into an Organization Account requires that you assign a new user as the owner of the Organization account. You will have the opportunity to sign up for a new Personal Account during the account conversion process and assign it or select an existing Personal Account (with a different username) to be the primary Owner of the Organization.

Reviewer 2: ```suggestion
The Organization Account will adopt your old username. Converting a Personal Account into an Organization Account requires that you designate a new user as the owner of the Organization account. You will have the opportunity to designate an existing user or sign up as a new user during the account conversion process.
```

The example below is Step 3 in the conversion process on a Personal Account with the username "kamalaya8":

<ImageSpotlight alt="Example demonstrating the Step 3 of the conversion process." src="/static/images/accounts/converting-personal-account.jpg" style={{maxWidth: 720}} />

> After completing the conversion process, you can no longer access the Personal Account you converted. To continue using Expo services, you will log into [expo.dev](https://expo.dev/) with the Personal Account you selected or created to manage the Organization Account. Then, select the organization from the top left dropdown to access the organization.

Reviewer 2: ```suggestion
After completing the conversion process, you can no longer log in as the old Personal Account's user. To continue using Expo services, you will log into [expo.dev](https://expo.dev/) with the user you selected or created to manage the Organization Account. Then, select the organization from the top left dropdown to access the organization.
```

> > **Note**: Before account conversion, make sure that all of your projects are on Expo SDK 43 or above. If not, you must upgrade it and rebuild it for notifications to continue to work on those projects.

Reviewer 2: ```suggestion
```

(this is only true for account renaming and project transfer)

### Renaming an Account

> Accounts can be renamed a limited number of times. Members in an Organization with Owners or Admin roles can rename the Account. Visit [the account settings](https://expo.dev/accounts/[account]/settings) and follow the steps under [Rename Account](https://expo.dev/settings#rename-account).

Reviewer 2: ```suggestion
Accounts can be renamed a limited number of times. Only Owners may rename accounts. Visit [the account settings](https://expo.dev/accounts/[account]/settings) and follow the steps under [Rename Account](https://expo.dev/settings#rename-account).
```

<ImageSpotlight alt="Rename Account settings panel." src="/static/images/accounts/rename-account.jpg" style={{maxWidth: 720}} />

#### Caveats

> Before renaming an account, make sure that your project is on Expo SDK 43 or above. If not, you must upgrade it and rebuild it for OTA updates to continue to work on that project.

Reviewer 2: ```suggestion
After renaming an account, new publishes to projects within the account must be on Expo SDK 43 or above.
```

### Transferring projects between Accounts

Projects can be transferred a limited number of times. A user must be an Owner or Admin on both source and destination accounts to transfer projects between them. Visit [the project settings page](https://expo.dev/accounts/[account]/projects/[project]/settings) and follow the steps under Transfer project.

<ImageSpotlight alt="Transfer Projects settings panel." src="/static/images/accounts/transfer-project.jpg" style={{maxWidth: 720}} />

#### Caveats

> Before transferring an account, make sure that your project is on Expo SDK 43 or above. If not, you must upgrade it and rebuild it for OTA updates to continue to work on that project.

Reviewer 2: ```suggestion
Before transferring a project, make sure that your it is on Expo SDK 43 or above. If not, you must upgrade it and rebuild it for OTA updates to continue to work on that project.
```
